src/components/dashboard/SignSettings.tsx

```diff
@@ -1,95 +1,87 @@
-      }
-    }
+  const [signatureAllowanceEnabled, setSignatureAllowanceEnabled] = useStorage(
+    {
+      key: "signatureAllowanceEnabled",
+      instance: ExtensionStorage
+    },
+    true
+  );
 
-    initializeSettings();
+  const handleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    valueChanged.current = true;
+    setEditingValue(e.target.value);
   }, []);
 
-  const toggleSignSettings = async () => {
-    const newSetting = !signSettingsState;
-    setSignSettingsState(newSetting);
-    await ExtensionStorage.set("setting_sign_notification", newSetting);
-  };
+  const handleBlur = useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      const newAllowance = Number(e.target.value);
 
-  const handleBlur = async (e) => {
-    const newAllowance = Number(e.target.value);
-    if (newAllowance !== signatureAllowance) {
-      trackEvent(EventType.SEND_ALLOWANCE_CHANGE, {
-        before: signatureAllowance,
-        after: newAllowance
-      });
-      setSignatureAllowance(newAllowance);
+      if (newAllowance !== signatureAllowance) {
+        trackEvent(EventType.SEND_ALLOWANCE_CHANGE, {
+          before: signatureAllowance,
+          after: newAllowance
+        });
+        setSignatureAllowance(newAllowance);
+      }
+    },
+    [signatureAllowance]
+  );
 
-      // Save the updated allowance to the extension storage
-      await ExtensionStorage.set("signatureAllowance", newAllowance);
+  useEffect(() => {
+    if (!valueChanged.current) {
+      setEditingValue(signatureAllowance);
     }
-  };
-
-  const handleChange = (e) => {
-    setEditingValue(e.target.value);
-  };
+  }, [signatureAllowance]);
 
   return (
-    <>
-      <Wrapper>
-        <PermissionCheckbox
-          checked={signSettingsState}
-          onChange={toggleSignSettings}
-        >
-          {browser.i18n.getMessage(
-            !!signSettingsState ? "enabled" : "disabled"
-          )}
-          <br />
-          <Text noMargin>
-            {browser.i18n.getMessage("setting_sign_notification_description")}
-          </Text>
-        </PermissionCheckbox>
-        <Spacer y={1.7} />
-        <InputV2
-          label={browser.i18n.getMessage("password_allowance")}
-          type="number"
-          value={editingValue}
-          onChange={handleChange}
-          onBlur={handleBlur}
-          fullWidth
+    <Wrapper>
+      <ToggleSwitchWrapper>
+        <Text>{browser.i18n.getMessage("enable_transfer_settings")}</Text>
+        <ToggleSwitch
+          checked={signatureAllowanceEnabled}
+          setChecked={setSignatureAllowanceEnabled}
         />
-      </Wrapper>
-    </>
+      </ToggleSwitchWrapper>
+      <InputV2
+        label={browser.i18n.getMessage("password_allowance")}
+        type="number"
+        value={editingValue}
+        onChange={handleChange}
+        onBlur={handleBlur}
+        disabled={!signatureAllowanceEnabled}
+        fullWidth
+      />
+    </Wrapper>
```

Comment:
Wdyt about hiding the input if the transfer setting is disabled?

Reply:
Its disabled at the moment but we can hide it as well. Let me push a new commit for it.